### PR TITLE
Dv batch

### PR DESCRIPTION
**Pull request by dianavermilya**
- checks if data is a list to determine if batch or single request
- removes support for list of list pixel data as image representation
- rewrites tests to use test images rather than list of list pixel data
@madisonmay

### CODE REVIEW

File: indicoio/utils/api.py [hunk 1]
```diff
@@ -2,18 +2,25 @@
...
 from indicoio import JSON_HEADERS
 from indicoio import config
 
-def api_handler(arg, cloud, api, url_params = {"batch":False, "api_key":None}, **kwargs):
+def api_handler(arg, cloud, api, url_params=None, **kwargs):
+    if url_params is None:
+        url_params = {"api_key":None}
+    elif url_params['batch'] == True: # image passed in is a list and not np.array
+        warnings.warn(
+            "Requests to 'batch_' + API name will be deprecated in the next major update.  Please send single and batch requests to the API name.",
```
Comment (dianavermilya) [inline indicoio/utils/api.py]: What do you think of the warning.  Also, is this the right place for the message?

Comment (sihrc) [inline indicoio/utils/api.py]: I think it makes most sense to put it where we create the partials for batch functionality. (so where it allows people to call batch_api()). probably wrap that partial in a function that throws a warning.

File: indicoio/__init__.py [hunk 2]
```diff
@@ -23,9 +24,18 @@
 
 from indicoio.config import API_NAMES
 
+def deprecation_decorator(f, api):
+    @wraps(f)
+    def wrapper(*args, **kwds):
+        warnings.warn(
+            "'batch_" + api + "' will be deprecated in the next major update.  Please call '" + api + "' instead.",
```
Comment (madisonmay) [inline indicoio/__init__.py]: Maybe slight wording change here, making it clear that you send the exact same arguments over.

**Comment by madisonmay** [comment]
Looks good to me!  Still think we need to make a slight edit to the warning, but other than that looks great!

File: indicoio/__init__.py (hunk 2 above)
Comment (dianavermilya) [inline indicoio/__init__.py]: Ok, I changed it to say: 
Please call '" + api + "' instead with the same arguments."
I can add 'exact' if you think that'd be better.

Comment (dianavermilya) [inline indicoio/__init__.py]: Cool.  I updated it.

**Comment by madisonmay** [comment]
Everything looks good to me!  Let's merge this into development, but I'm going to hold off on merging to master and pushing to PyPI until we can update the docs.

**Comment by madisonmay** [comment]
Whoops, just realized we're missing one thing -- tests to make sure that batch calls sent to the non-batch functions work properly.  Could you add some basic tests in before merge?

**Comment by dianavermilya** [comment]
@madisonmay ok this is ready for another look.

File: indicoio/__init__.py [hunk 3]
```diff
@@ -23,9 +24,26 @@
...
+            DeprecationWarning
+        )
+        return f(*args, **kwargs)
+    return wrapper
 
+def detect_batch_decorator(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        if isinstance(args[0], list):
+            kwargs['batch']=True
```
Comment (madisonmay) [inline indicoio/__init__.py]: <nitpicking>
```
kwargs['batch'] = True
```
</nitpicking>

**Comment by madisonmay** [comment]
I think this is good to go, I"ll merge after rebase!

**Comment by madisonmay** [comment]
Build passed, merging to development.

**Pull request merged**